stlab/forest_algorithms.hpp

```diff
@@ -5,16 +5,18 @@
 
 /**************************************************************************************************/
 
-// stdc++
-#include <optional>
-
 // stlab
+#include <stlab/concurrency/optional.hpp>
 #include <stlab/forest.hpp>
 
 /**************************************************************************************************/
 
 namespace stlab {
 
+/**************************************************************************************************/
+
+namespace forests {
```

Comment:
If we are going to nest namespaces. Use a singular (`forest`).

Reply:
I cannot have both the class `stlab::forest<T>` and the namespace `stlab::forest`.

Reply:
No, I'd prefer not nesting... Or if these are very forest specific adding them to the class forest? (Similar to the std::list<> algorithms).